src/main.rs

```diff
@@ -75,10 +94,26 @@ async fn main()-> Result<(), Box<dyn Error>>{
 
 
 
-    let _ = tracing_subscriber::fmt()
+   // let _ = tracing_subscriber::fmt()
+   //     .json()
+   //     .with_env_filter(EnvFilter::new("debug"))
+   //     .try_init();
+    //
+    //
+    //
+    let node_span = debug_span!("libp2p_gossipsub::behaviour", node_id  = node_id);
+    let _entered = node_span.enter(); // keep the span entered
+
+//    let node_id_layer = writer::NodeIdFieldLayer::<Registry>::new(node_id);
+
+    let subscriber = Registry::default()
+        .with(fmt::Layer::default()
         .json()
-        .with_env_filter(EnvFilter::new("debug"))
-        .try_init();
+        .with_current_span(true))
+        .with(EnvFilter::new("debug"));
+
+    tracing::subscriber::set_global_default(subscriber).expect("failed to set global subscriber");
```

Comment:
i think we'll need to do the subscriber setup before creating & entering span to get the spans working properly
```suggestion
    let subscriber = Registry::default()
        .with(fmt::Layer::default()
        .json()
        .with_current_span(true))
        .with(EnvFilter::new("debug"));

    tracing::subscriber::set_global_default(subscriber).expect("failed to set global subscriber");

    let node_span = debug_span!("libp2p_gossipsub::behaviour", node_id  = node_id);
    let _entered = node_span.enter(); // keep the span entered
```